contracts/ReverseRegistrar.sol

```diff
@@ -1,31 +1,117 @@
-pragma solidity ^0.4.0;
+pragma solidity ^0.4.10;
 
-import './AbstractENS.sol';
+import "AbstractENS.sol";
+
+contract Resolver {
+    function setName(bytes32 node, string name) public;
+}
+
+/**
+ * @dev Provides a default implementation of a resolver for reverse records,
+ * which permits only the owner to update it.
+ */
+contract DefaultReverseResolver is Resolver {
+    // namehash('addr.reverse')
+    bytes32 constant ADDR_REVERSE_NODE = 0x91d1777781884d03a6757a803996e38de2a42967fb37eeaca72729271025a9e2;
+
+    AbstractENS ens;
+    mapping(bytes32=>string) public name;
+    
+    /**
+     * @dev Constructor
+     * @param ensAddr The address of the ENS registry.
+     */
+    function DefaultReverseResolver(address ensAddr) {
```

Comment:
You can already use `AbstractENS` as the type here instead of address, then you do not need an explicit conversion.